arkouda/dataframe.py

```diff
@@ -753,75 +758,280 @@ def __delitem__(self, key):
             self._empty = True
         self.update_nrows()
 
-    def __getitem__(self, key):
-        # convert series to underlying values
-        # Should check for index alignment
+    def validate_key(self, key):
         if isinstance(key, Series):
-            key = key.values
+            # TODO: check index alignment
+            return self.validate_key(key.values)
+        if isinstance(key, list):
+            return self.validate_key(array(key))
+
+        if isinstance(key, slice):
+            if key.start is not None and key.start < 0:
+                raise ValueError("Negative start index not supported")
+            if key.stop is not None and key.stop > len(self):
+                raise ValueError("Slice stop index out of range")
+            return key
```

Comment:
I think just distance between the implementations. I'll do the same here for uniformity

Reply:
On second thought, it's because slice indexing behavior is different from array indexing behavior for dataframes. slices do rows while arrays do columns. This maintains that difference so it can be used to distinguish in the body of `__getitem__`